### PR TITLE
Add more Chapel configurations to linux packages

### PR DESCRIPTION
**Pull request by jabraham17**
Improves the chapel linux packages to be more "all batteries included". This means we will have a single RPM for each platform which contains COMM=none, COMM=gasnet, and COMM=ofi builds for multiple configurations, as well as a CHPL_GPU=cpu build

other improvements
- Be explicit about many defaults like `CHPL_TARGET_COMPILER` and `CHPL_TARGET_CPU`
- Pin to specific versions of compilers where applicable, for example `CHPL_LLVM_CONFIG`
- lay groundwork for using more system packages, which decreases binary size and improves package quality

This PR depends on https://github.com/chapel-lang/chapel/pull/25251.
Resolves https://github.com/chapel-lang/chapel/issues/27331

Testing
- [x] Spot checked packages to make sure everything still works
  - [x] tested ubuntu24
  - [x] tested debian 12
  - [x] tested fedora 42
  - [x] tested el9

[Reviewed by @arifthpe]

### CODE REVIEW

File: util/packaging/common/fill_docker_template_common.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+default_config = {
+    "CHPL_HOST_PLATFORM": "linux64",
+    "CHPL_TARGET_PLATFORM": "linux64",
+    "CHPL_TARGET_COMPILER": ["llvm", "gnu", "clang"],
+    "CHPL_TARGET_CPU": "none",
+    "CHPL_LOCALE_MODEL": "flat",
+    "CHPL_COMM": "none",
+    "CHPL_TASKS": "qthreads",
+    "CHPL_LAUNCHER": [None, "slurm-srun"],
```
Comment (arifthpe) [inline util/packaging/common/fill_docker_template_common.py:83]: how come this (and other `CHPL_LAUNCHER` entries) is the only `None` and elsewhere we use string `"none"`?

Comment (jabraham17) [reply util/packaging/common/fill_docker_template_common.py:83]: In this case, `None` means unset and `"none"` means set to the literal value "none".

This is done because by leaving CHPL_LAUNCHER unset, we allow chplenv to infer the right values for variables. Otherwise what happens is a user does `--comm=gasnet`, which defaults to udp. Which results in a GASNet UDP executable with no amudprun launcher, which has confusing errors for users. Unfortunately, the compiler will happily build COMM=gasnet with UDP and no launcher.

TL;DR, in this instance, inferring the value of CHPL_LAUNCHER is advantageous
